src/main/java/edu/kit/datamanager/repo/util/validators/impl/HandleNetValidator.java

```diff
@@ -0,0 +1,156 @@
+        Pattern pattern = Pattern.compile(regex, Pattern.MULTILINE);
+        Matcher matcher = pattern.matcher(input);
+        if (matcher.find()) {
+            if (matcher.group(1).equals("https://") || matcher.group(1).equals("http://"))
+                return isValidHTTPURL(matcher.group(0));
+            else return isValidHandle(matcher.group(2));
+        } else return isValidHandle(input);
+    }
+
+    /**
+     * This method checks if the given prefix and suffix is downloadable.
+     * If there is some invalid input, it will validate the prefix with help of the RESTful API by handle.net.
+     *
+     * @param serverAddress from the server which should be uses for the validation.
+     * @param prefix        the handle prefix
+     * @param suffix        the handle suffix
+     * @return true if the record is downloadable
+     */
+    private boolean isDownloadable(String serverAddress, String prefix, String suffix) {
+        IValidator urlValidator = new URLValidator();
+        AtomicBoolean fullValid = new AtomicBoolean(false);
+        log.debug("Server address: {}", serverAddress);
+        log.debug("Prefix: {}", prefix);
+        log.debug("Suffix: {}", suffix);
+        ignoringExc(() -> {
+            try{
+                fullValid.set(urlValidator.isValid(serverAddress + "/" + prefix + "/" + suffix));
+            } catch (Exception ignored){
+            }
+        });
+        if (fullValid.get()) {
+            LOG.info("The handle is valid!");
+            return true;
+        }
+        log.warn("Either the suffix or the prefix might be invalid. Proving if prefix is valid...");
+        if (urlValidator.isValid("https://hdl.handle.net/0.NA/" + prefix)) {
+            log.info("The prefix {} is valid!", prefix);
+            throw new MessageValidationException("Prefix valid, but suffix not");
+        }
+        log.error("The entered prefix is invalid!");
+        throw new BadArgumentException("Prefix not provable on handle.net");
+    }
+
+    /**
+     * Uses the other isDownloadable method with the handle.net server as serverAddress
+     *
+     * @param prefix the handle prefix
+     * @param suffix the handle suffix
+     * @return true if the record is downloadable
+     */
+    private boolean isDownloadable(String prefix, String suffix) {
+        return isDownloadable("http://hdl.handle.net/api/handles", prefix, suffix);
+    }
+
+    /**
+     * This method uses regex to get the prefix, suffix and the server address out of a http or https URL.
+     * After this it uses the isDownloadable method to check whether the input is downloadable.
+     *
+     * @param url to validate
+     * @return true if the input is valid and downloadable.
+     */
+    private boolean isValidHTTPURL(String url) {
+        String regex = "(http|https)://(.+)/([A-Za-z0-9.]+)/([A-Za-z0-9.]+)";
+        Pattern pattern = Pattern.compile(regex, Pattern.MULTILINE);
```

Comment:
Give me a hammer and everything looks like a nail!?
Why not using URL class to validate URL?